### PR TITLE
Describe `matrix:` URI scheme

### PR DESCRIPTION
**Pull request by turt2live**
Replaces https://github.com/matrix-org/matrix-doc/pull/3168
MSC: https://github.com/matrix-org/matrix-doc/pull/2312

Updated for modern times.

Changelog entry compatible with https://github.com/matrix-org/matrix-doc/pull/3609

Preview: https://pr3608--matrix-org-previews.netlify.app

### CODE REVIEW

File: content/appendices.md [hunk 1]
```diff
@@ -656,24 +656,136 @@ homeserver to look up the alias.
...
+nonid-segment = segment-nz ; as defined in RFC 3986
+type-qualifier = segment-nz "/" ; as defined in RFC 3986
+id-without-sigil = string ; as defined in Matrix identifier spec above
+query = query-element *( "&" query-item )
+query-item = action / routing / custom-query-item
+action = "action=" ( "join" / "chat" )
+routing = "via=” authority
+custom-query-item = custom-item-name "=" custom-item-value
+custom-item-name = 1*unreserved ; reverse-DNS name
+custom-item-value =
```
Comment (ShadowJonathan) [inline content/appendices.md:696]:
```suggestion
custom-item-value = string
```
?

File: content/appendices.md [hunk 2]
```diff
@@ -656,24 +656,136 @@ homeserver to look up the alias.
...
+```
+
+Note that this format is deliberately following [RFC 3986](https://tools.ietf.org/html/rfc3986)
+to ensure maximum compatibility with existing tooling. The scheme name (`matrix`) is
+registered alongside other schemes by the IANA [here](https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml).
+
+Currently, the `authority` and `fragment` are unused by this specification,
+though are reserved for future use. Matrix does not have a central authority which
+could reasonably fill the `authority` role. `nonid-segment` in the schema is
```
Comment (ShadowJonathan) [inline content/appendices.md:705]: I think some explanation of what the `nonid-segment` could possibly be(come) would help. I'm not getting a sense of understanding what it is while reading this.

File: content/appendices.md [hunk 3]
```diff
@@ -656,24 +656,136 @@ homeserver to look up the alias.
...
+    by the URI and thus is only valid on URIs which are referring to a room (it
+    has no meaning and is ignored otherwise). The client should prompt for confirmation
+    prior to joining the room, if the user isn't already part of the room.
+  * `action=chat` - Describes an intent for the client to start/open a DM with
+    the user described by the URI and thus is only valid on URIs which are referring
+    to a user (it has no meaning and is ignored otherwise). Clients supporting a
+    form of Canonical DMs should reuse existing DMs instead of creating new ones
+    if available. The client should prompt for confirmation prior to creating the
+    DM, if the user isn't being redirected to an existing canonical DM.
+* `via` - Can be used to denote which servers (authorities) to attempt to resolve
```
Comment (ShadowJonathan) [inline content/appendices.md:746]: I think this somewhat clashes with the rejection of "authority" just a few paragraphs prior, so maybe leaving it out would avoid confusion a bit better.

File: content/appendices.md [hunk 4]
```diff
@@ -656,24 +656,136 @@ homeserver to look up the alias.
...
+  the resource through, or take `action` through. An example of using `via` for
+  routing Room IDs is described [below](#routing), and is encouraged for use in
+  Matrix URIs referring to a room ID. Matrix URIs can additionally use this `via`
+  parameter for non-public federation resolution of identifiers (i.e.: listing a
+  server which might have information about the given user) while a more comprehensive
+  way is being worked out, such as one proposed by [MSC3020](https://github.com/matrix-org/matrix-doc/pull/3020).
```
Comment (ShadowJonathan) [inline content/appendices.md:752]: Should we really link to in-progress MSCs in the spec? Is there precedent for this?

Alternatively, simply noting that a more comprehensive solution is worked out would work better here, and anyone asking what it is in #spec could then get redirected to the up-to-date work.

File: content/appendices.md (hunk 1 above)
Comment (turt2live) [reply content/appendices.md:696]: It can be anything permitted by the RFC. This is described later on.

File: content/appendices.md (hunk 2 above)
Comment (turt2live) [reply content/appendices.md:705]: The MSC doesn't appear to have much more information, sorry.

File: content/appendices.md (hunk 4 above)
Comment (turt2live) [reply content/appendices.md:752]: We do this when it adds context and clarity that isn't able to be captured by the spec. The language deliberately points to the MSC in vague ways ("such as", "could be implemented like", "similar to", etc) in order to ensure we don't seal the MSC's fate into the spec.

If a different solution comes along, great :)

File: content/appendices.md (hunk 3 above)
Comment (turt2live) [reply content/appendices.md:746]: This is trying to bring the grammar of an `authority` through, but it should probably just say that.

File: content/appendices.md (hunk 2 above)
Comment (KitsuneRal) [reply content/appendices.md:705]: I can't really define it beyond "a segment to specify a resource that is not supposed to have an id", which is pretty tautological and not any clearer. The idea was that some resources don't need an id but could benefit from having a URI - as a contrived example, some kind of a cloak-room/homepage for a space. 

In any case, this PR is meant for purely technical purposes so not a good place to introduce new wording; and as Travis said, the original MSC doesn't have anything to build on, anyway.

**Review by KitsuneRal — APPROVED** [review]
Thank you!

**Pull request merged**